### PR TITLE
fix(jira): Fall back to other projects if project was deleted

### PR DESCRIPTION
**Pull request by ceorourke**
If a user manually creates a Jira ticket through Sentry, we save the project key as a default setting (so when you open the issue creation modal on another issue, it is pre-populated) and cache the metadata for faster future issue creation. However, this becomes a problem if the Jira project is deleted. This PR first attempts to get the metadata of the default project id, but if that fails, it makes an attempt to get metadata from the next project in a list of all projects for that Jira instance.

### CODE REVIEW

File: src/sentry/integrations/jira/integration.py [hunk 1]
```diff
@@ -506,42 +506,43 @@ def get_issue_type_meta(self, issue_type, meta):
...
-                "attempts": attempts,
-                "jira_projects": jira_project_ids,
-            },
-        )
-        raise IntegrationError(
-            "Could not get issue create metadata for any Jira projects. "
-            "Ensure that your project permissions are correct."
-        )
+            meta = self.fetch_issue_create_meta(client, project_id)
+        if not meta:
```
Comment (AniketDas-Tekky) [inline src/sentry/integrations/jira/integration.py:512]: Should this be `meta is None`? an empty valid meta object would enter this if-block

**Review by AniketDas-Tekky — COMMENTED** [review]
Can you add a test?

File: tests/sentry/integrations/jira/test_integration.py [hunk 2]
```diff
@@ -330,6 +331,40 @@ def test_get_create_issue_config_with_default(self):
...
+        )
+        group = event.group
+        installation = self.integration.get_installation(self.organization.id)
+        installation.org_integration.config = {
+            "project_issue_defaults": {str(group.project_id): {"project": "10004"}}
+        }
+        installation.org_integration.save()
+
+        with mock.patch.object(installation, "get_client", get_client):
+            mock_get_issue_create_meta.return_value = json.loads(
```
Comment (ceorourke) [inline tests/sentry/integrations/jira/test_integration.py:353]: idk if I'm doing this test right :( I really struggled with this.

File: src/sentry/integrations/jira/integration.py [hunk 3]
```diff
@@ -506,42 +506,45 @@ def get_issue_type_meta(self, issue_type, meta):
...
-            },
-        )
-        raise IntegrationError(
-            "Could not get issue create metadata for any Jira projects. "
-            "Ensure that your project permissions are correct."
-        )
+            meta = self.fetch_issue_create_meta(client, project_id)
+        if meta is not None:
+            return meta
+        if meta is None:
```
Comment (AniketDas-Tekky) [inline src/sentry/integrations/jira/integration.py:514]: Don't need this `if` statement as the only way we'll be here is if `meta` is `None`

Comment (ceorourke) [reply src/sentry/integrations/jira/integration.py:514]: oh yeah oops lol

**Review by AniketDas-Tekky — APPROVED** [review]

**Pull request merged**